### DISCUSSION TITLE
Compatibility with Visual Micro

### DISCUSSION THREAD

**Discussion by AlexandrJet** [stm32duino/Arduino_Core_STM32#2270] [Q&A] [answered]
Hi everyone. After upgrading to 2.7.1, any project for STM32 is no longer compiled. I noticed that before that it was necessary to install compilers for STM32 myself. Now, as I understand it, all compilers are included.
Can I find out which version, starting from 2.4.0 (everything worked on it for sure), was installed the old way before that?
I have attached the error log during compilation below.

Logs:
>Compiling 'MyProject' for 'Generic STM32F4 series'
>Error compiling variant
>arm-none-eabi-gcc*: warning: {compiler.optimization_flags}: linker input file unused because linking not done
>Build failed for project 'MyProject'
>arm-none-eabi-gcc*: error: {compiler.optimization_flags}: linker input file not found: No such file or directory
>arm-none-eabi-gcc*: warning: {compiler.optimization_flags}: linker input file unused because linking not done
>arm-none-eabi-gcc*: error: {compiler.optimization_flags}: linker input file not found: No such file or directory

This happens now with any project for Stm32. I think, it can be seen from the log that the problem is in choosing the optimization flag, I tried to change the optimization, but it did not help. Reinstalling Visual Micro and Arduino_Core_STM32 did not help.
Everything works in the Arduino IDE.

PS: Sorry, for my English.

**Comment by fpistm** [comment, marked as answer]
Hi @AlexandrJet 
The toolchain was always provided within the core.
The change which impact your setup is probably this:
https://github.com/stm32duino/Arduino_Core_STM32/pull/2184

Unfortunately, could do nothing as it is Arduino specification.
